src/reducers/user.js

```diff
@@ -0,0 +1,43 @@
+import asyncStore from "utils/asyncstorage";
+import userServices from "services/user";
+import setUserError from "./error";
+
+export const SET_USER = "SET_USER";
+export const CLEAR_USER = "CLEAR_USER";
+
+const initialState = {};
+
+export default function userReducer(state = initialState, action) {
+  switch (action.type) {
+    case SET_USER:
+      return action.payload;
```

Comment:
Aren't reducers normally supposed to return state with updated values from the action payload? So something like `{...state, user: action.payload}`. That's how I see it working in the `errorsReducer`.

Reply:
In the error reducer, there's a different error for each type (a user error, an event error, etc.), so when we're only updating one of those errors, we need to define the state as a clone of the current state which includes all errors, plus the updated version of the error we're changing. The user reducer is a little bit different in that it's just one thing we need to keep track of-- when you set a user, you're setting the user state to whatever the payload is (which I believe is a login object), and when you clear it you clear the entire state because there's only the one user in it.